src/renderer/vt/VtSequences.cpp

```diff
@@ -266,15 +262,15 @@ using namespace Microsoft::Console::Render;
 [[nodiscard]] HRESULT VtEngine::_SetGraphicsRenditionRGBColor(const COLORREF color,
                                                               const bool fIsForeground) noexcept
 {
-    const std::string fmt = fIsForeground ?
-                                "\x1b[38;2;%d;%d;%dm" :
-                                "\x1b[48;2;%d;%d;%dm";
-
     DWORD const r = GetRValue(color);
     DWORD const g = GetGValue(color);
     DWORD const b = GetBValue(color);
 
-    return _WriteFormattedString(&fmt, r, g, b);
+    // Worst case scenario: "\x1b[38;2;128;128;128m", which has length = 19 + \0.
+    // Use small_vector as MSVC's SSO only buffers up to 15 characters.
+    boost::container::small_vector<char, 20u> buf;
+    const auto end = fmt::format_to(std::begin(buf), FMT_COMPILE("\x1b[{};2;{};{};{}m"), fIsForeground ? 38 : 48, r, g, b);
+    return _Write({ buf.data(), static_cast<size_t>(end - buf.begin()) });

```

Comment:
Small vector is good. The formatting is just as fast as `char[N]`. But converting it to `std::string_view` seems slower, due to the using of `static_cast<size_t>(end - buf.begin())`. 

Note that I can't just use `{ buf.data(), buf.size() }`, because the formatted string may not take all the spaces in the buffer.

On that note, do I still need to add the space for '\0'. I mean std::string does *not* require a string to end with '\0', right?

Reply:
buf.size() should work actually since the container keeps track of its own size.

Reply:
Yeah you're right. I was using `std::begin()`. It should be `std::back_inserter`, then `buf.size()` is good.

Reply:
Why not
```cpp
fmt::basic_memory_buffer<char, 20> buf;
fmt::format_to(buf, FMT_COMPILE("\x1b[{};2;{};{};{}m"), fIsForeground ? 38 : 48, r, g, b);
buf.append('\0');
return _Write({ buf.data(), buf.size() - 1 });
```

Reply:
Edit: removed unneeded null terminator and clarified a bit wrt std::string and std::string_view's respective handling of null terminators

Reply:
OMG There's a `fmt::memory_buffer` and `fmt::wmemory_buffer` class! Thanks @sylveon! 😍

Reply:
FYI: `fmt::basic_memory_buffer` and its derivatives don't need a `std::back_inserter`.
The `fmt::format_to` function appears to have template specializations for `fmt::basic_memory_buffer`.

Reply:
I tried this but it does not work. The signature is:


```cpp

template <typename S, typename... Args, size_t SIZE = inline_buffer_size,
          typename Char = enable_if_t<detail::is_string<S>::value, char_t<S>>>
inline typename buffer_context<Char>::iterator format_to(
    basic_memory_buffer<Char, SIZE>& buf, const S& format_str, Args&&... args) {
}
```

This seems to work only is `S` is valid when checked against `detail::is_string<S>`

Reply:
The specializations for `fmt::basic_memory_buffer` are apparently being removed from `fmt` anyways.
Instead it'll feature `fmt::appender` in the next version, which would then also work with `fmt::format_to`.

Reply:
I'm more than happy with the current solution. It's elegant. It's fast. And it's also maintainable. Now I'm confident this is good enough. Thanks everyone who contributed to this thread. You guys are awesome!